Scenes/Player Scripts/Player.gd

```diff
@@ -0,0 +1,319 @@
+		move_and_slide(cur_velocity) # do not multiply delta by velocity
+		rotate(rot_velocity * delta) # do multiply delta by velocity
+	else:
+		rotation = lerp_angle(rotation, puppet_player_rotation, delta * 8)
+		if not tween.is_active():
+			move_and_slide(puppet_player_velocity)
+
+func _on_body_entered(body: Node):
+	if body is RigidBody2D:
+		rpc("_on_player_died")
+
+func _on_difficulty_change(sides):
+	cur_sides = sides
+	print("changing shape")
+	_change_shape(cur_sides)
+
+remotesync func _on_player_died():
+	username_instance.player_following = null
+	if username_instance != null:
+		username_instance.visible = false
+	hide()
+	yield(get_tree().create_timer(1), "timeout")
+	queue_free() # delete this player
+	# do whatever necessary to show the player died here
+
+func get_sides() -> int:
+	return cur_sides
+
+func _apply_points(sides: int, point_sets: Array, polygon_points: PoolVector2Array):
+	assert(sides == point_sets.size())
+	var id := 0
+	# reassign existing points if they exist
+	while id < line_segments.size():
+		line_segments[id].set_position(point_sets[id], outline_width)
+		id += 1
+	# create new points if needed
+	while id < point_sets.size():
+		line_segments.append(LineSegment.new())
+		line_segments[-1].setup(color)
+		line_segments[-1].set_position(point_sets[id], outline_width)
+		add_child(line_segments[-1].line_2d)
+		add_child(line_segments[-1].collision_shape_2d)
+		id += 1
+	# assign points to inner boundary polygon
+	get_node("InnerBoundary").get_node("CollisionShape").polygon = polygon_points
+	get_node("InnerBoundary").get_node("VisibleShape").polygon = polygon_points
+
+
+puppet func _change_hidden_sides(hidden_sides: int, network_sides: Array):
+	if get_tree().get_network_unique_id() == 1:
+		# should never be less than 2 visible sides (or player orientation is difficult to determine)
+		assert(line_segments.size() - hidden_sides >= 2)
+		for line_segment in line_segments:
+			line_segment.show()
+		var possible_hidden_sides = range(0, line_segments.size() - 1)
+		var side_ids = []
+		for _i in range(hidden_sides):
+			# same thing but shorter
+			# line_segments[possible_hidden_sides.pop_at(random.randi_range(0, possible_hidden_sides.size() - 1))].hide()
+			# choose random element from possible_hidden_sides
+			var id: int = random.randi_range(0, possible_hidden_sides.size() - 1)
+			# find the side id
+			var hidden_side_id: int = possible_hidden_sides[id]
+			# delete the id so it cannot be hidden more than once
+			possible_hidden_sides.pop_at(id)
+			# hide the segment
+			side_ids.append(hidden_side_id)
+			line_segments[hidden_side_id].hide()
+		rpc("client_change_hidden_sides", side_ids)
+
+remote func client_change_hidden_sides(network_sides: Array):
+	for line_segment in line_segments:
+		line_segment.show()
+	
+	for _i in network_sides:
+		var id: int = _i
+		# hide the segment
+		line_segments[id].hide()
+
+func _increase_missing_sides():
+	if line_segments.size() - (max_missing_sides + 1) >= 2:
```

Comment:
Simplify the math?  Unless you have the redundant math to read like how you're thinking about the logic... in that case just make a function that describes it. 
```suggestion
	if line_segments.size() - max_missing_sides >= 1:
```